utils/riff/recommendations/rec_base.js

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) 2018-present Riff Learning, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {MAX_REC_DISPLAY_NUMBER} from './rec_constants';
+import {getWeekNumber} from './time';
+
+/**
+ * internal contains all module internal values used by the exported
+ * classes/functions defined in this module. This provides a way to
+ * mock these values in unit tests of the exported functions.
+ */
+const internal = {
+    getWeekNumber,
+    MAX_REC_DISPLAY_NUMBER,
+};
+
+/* ******************************************************************************
+ * RecommendationBase                                                      */ /**
+ *
+ * Abstract base class for all Recommendations.
+ * A Recommendation describes some action the user should take because Riff
+ * believes it will enhance their performance (e.g. make the meetings they attend
+ * more productive, help them do better in a course...) based on information known
+ * at the current time (such as if the user has had a riff video meeting, or how
+ * many interactions in various channels they've had).
+ *
+ ********************************************************************************/
+class RecommendationBase {
+    /* **************************************************************************
+     * constructor                                                         */ /**
+     *
+     * RecommendationBase class constructor.
+     *
+     * @param {!RecommendationBase.Config} config
+     *      The settings to configure this RecommendationBase
+     */
+    constructor(config) {
+        // instance properties
+
+        /** Unique recommendation type identifier
+         *  @type {string}
+         */
+        this.recType = config.recType;
```

Comment:
It might be nice to document, somewhere in the class, that recTypes are defined as constants in `rec_constants`.